src/kubernetes_cluster/spec/kubernetes_api/state_machine.rs

```diff
@@ -199,22 +228,25 @@ pub open spec fn handle_update_request(msg: Message, s: KubernetesAPIState) -> (
 {
     let req = msg.content.get_update_request();
     if Self::validate_update_request(req, s).is_Some() {
+        // Update fails.
         let result = Err(Self::validate_update_request(req, s).get_Some_0());
         let resp = form_update_resp_msg(msg, result);
         (s, resp)
     } else if Self::update_is_noop(req.obj, s.resources[req.key]) {
         // Update is a noop because there is nothing to update
-        // so the resource version counter does not increase here
+        // so the resource version counter does not increase here.
         let result = Ok(s.resources[req.key]);
         let resp = form_update_resp_msg(msg, result);
         (s, resp)
     } else {
-        // Update succeeds
-        // Updates the resource version of the object
-        let updated_obj = req.obj.set_namespace(req.key.namespace).set_resource_version(s.resource_version_counter);
+        // Update succeeds.
+        // Updates the resource version of the object.
+        let old_obj = s.resources[req.key];
+        let updated_obj = req.obj.set_namespace(req.key.namespace)
+                            .set_resource_version(s.resource_version_counter)
+                            .set_uid(old_obj.metadata.uid.get_Some_0());
```

Comment:
Why do we have to set the `uid` if the old and new uid are required to be the same in `validate_update_request`?

Reply:
We only require the uid to be the same when `obj.metadata.uid.is_Some()`. If the uid is not present in the metadata, we will set it to the existing uid. See this: https://github.com/kubernetes/kubernetes/blob/v1.26.3/staging/src/k8s.io/apiserver/pkg/registry/rest/update.go#L133

Reply:
Didn't see `is_Some()`.